dask_cuml/linear_model/linear_regression.py

```diff
@@ -0,0 +1,560 @@
+        return build_host_dict(workers)
+
+
+def _fit_on_worker(data, params):
+    ipc_dev_list, devarrs_dev_list = data
+
+    # Open 1 ipc thread per device
+    open_ipcs = []
+    for p, dev in ipc_dev_list:
+        arrs = []
+        for x, y, coef in p:
+            arrs.append(x)
+            arrs.append(y)
+            arrs.append(coef)
+        ipct = new_ipc_thread(arrs, dev)
+        open_ipcs.append(ipct)
+
+    # alloc_info = list(itertools.chain([t.info() for t in open_ipcs]))
+    # alloc_info.extend(
+    #     list(itertools.chain(
+    #         [[(build_alloc_info(X)[0], build_alloc_info(y)[0],
+    #            build_alloc_info(coef)[0]) for X, y, coef in p]
+    #          for p, dev in devarrs_dev_list])))
+
+    alloc_info = []
+    for t in open_ipcs:
+        outsiders = t.info()
+        triplet = []
+        for i in range(0, len(outsiders), 3):
+            triplet.append(outsiders[i])
+            triplet.append(outsiders[i+1])
+            triplet.append(outsiders[i+2])
+            alloc_info.append(triplet)
+
+    for p, dev in devarrs_dev_list:
+        locals = []
+        for X, coef, pred in p:
+            locals.append(build_alloc_info(X)[0])
+            locals.append(build_alloc_info(coef)[0])
+            locals.append(build_alloc_info(pred)[0])
+        alloc_info.append(locals)
+
+    try:
+        from cuml.linear_model.linear_regression_mg import LinearRegressionMG as cuOLS
+        ols = cuOLS()
+        intercept = ols._fit_dask(alloc_info, params)
+    except Exception as e:
+        print("FAILURE in FIT: " + str(e))
+
+    [t.close() for t in open_ipcs]
+    [t.join() for t in open_ipcs]
+
+    return intercept
+
+
+def _predict_on_worker(data, intercept, params):
+    ipc_dev_list, devarrs_dev_list = data
+
+    open_ipcs = []
+    for p, dev in ipc_dev_list:
+        arrs = []
+        for mat, coef, pred in p:
+            arrs.append(mat)
+            arrs.append(coef)
+            arrs.append(pred)
+        ipct = new_ipc_thread(arrs, dev)
+        open_ipcs.append(ipct)
+
+    alloc_info = []
+    for t in open_ipcs:
+        outsiders = t.info()
+        triplet = []
+        for i in range(0, len(outsiders), 3):
+            triplet.append(outsiders[i])
+            triplet.append(outsiders[i+1])
+            triplet.append(outsiders[i+2])
+            alloc_info.append(triplet)
+
+    for p, dev in devarrs_dev_list:
+        locals = []
+        for X, y, coef in p:
+            locals.append(
+                build_alloc_info(X,unique=np.random.randint(0, 1e6))[0])
+            locals.append(
+                build_alloc_info(y, unique=np.random.randint(0, 1e6))[0])
+            locals.append(
+                build_alloc_info(coef, unique=np.random.randint(0, 1e6))[0])
+        alloc_info.append(locals)
+
+    try:
+        from cuml.linear_model.linear_regression_mg import LinearRegressionMG as cuOLS
+        ols = cuOLS()
+        ols._predict_dask(alloc_info, intercept, params)
```

Comment:
Maybe rename this to `predict_mg()` on the cuml side? I had done this so that we could decouple cuml from dask.

Reply:
For now there is no point, since the class structure in cuML is temporary while the multigpu distribution is sorted out. That said, it would be a small change for 0.7.